### PR TITLE
CONTENTLIB-2 email content library

### PR DESCRIPTION
**Pull request by seakchiew**
add new option to create content library for emails only with email toolbar and widgets in the richeditor

### CODE REVIEW

File: config/Config.cfc [hunk 1]
```diff
@@ -13,9 +14,13 @@ component {
 
 		if ( siteTreePos ) {
 			settings.adminSideBarItems.insertAt( siteTreePos+1, "contentLibrary" );
+			settings.adminSideBarItems.insertAt( siteTreePos+2, "emailContentLibrary" );
```
Comment (DominicWatson) [inline config/Config.cfc]: Could we use this same approach to add the alternative admin menu items?

The alt admin theme is now open source, though some/most pixl8 applications still using our internal one. Supporting the menu here will be the most portable way to do this.

File: config/Config.cfc [hunk 2]
```diff
@@ -32,4 +37,8 @@ component {
 
 	}
 
+		private void function _setupFeatures( required struct settings ) {
```
Comment (DominicWatson) [inline config/Config.cfc]: too many tabs (or using spaces here))

File: config/Config.cfc [hunk 3]
```diff
@@ -32,4 +37,8 @@ component {
...
+		settings.features.emailContentLibrary                 = { enabled=false , siteTemplates=[ "*" ], widgets=["email_content_library"] };
```
Comment (DominicWatson) [inline config/Config.cfc]: Lets enable this by default. Not sure any good reason not to.

File: forms/preside-objects/email_content_library_conditional_alternative.xml [hunk 4]
```diff
@@ -0,0 +1,10 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<form i18nBaseUri="preside-objects.content_library_conditional_alternative:">
```
Comment (DominicWatson) [inline forms/preside-objects/email_content_library_conditional_alternative.xml]: should use the feature flag on the form definition.

File: forms/preside-objects/email_content_library_conditional_alternative.xml [hunk 5]
```diff
@@ -0,0 +1,10 @@
...
+	<tab id="default">
+		<fieldset id="default" sortorder="10">
+			<field binding="content_library_conditional_alternative.content_library_content" control="hidden" />
```
Comment (DominicWatson) [inline forms/preside-objects/email_content_library_conditional_alternative.xml]: add sortorder="" to each field.

File: forms/preside-objects/email_content_library_content.xml [hunk 6]
```diff
@@ -0,0 +1,9 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<form i18nBaseUri="preside-objects.content_library_content:">
```
Comment (DominicWatson) [inline forms/preside-objects/email_content_library_content.xml]: should use the feature flag on the form definition

**Review by DominicWatson — CHANGES_REQUESTED** [review]
Nice one, looking good. Little bit of feedback

**Comment by seakchiew** [comment]
hi @DominicWatson 

I have made the changes and move the Email Content Library menu item ( sidebar and admin menu ) to Email Centre where it makes much more sense.

Thanks,
Seak Chiew

File: config/Config.cfc (hunk 2 above)
Comment (seakchiew) [reply config/Config.cfc]: fixed spacing issue

File: config/Config.cfc (hunk 1 above)
Comment (seakchiew) [reply config/Config.cfc]: moved the menu to emailCenter

File: forms/preside-objects/email_content_library_conditional_alternative.xml (hunk 4 above)
Comment (seakchiew) [reply forms/preside-objects/email_content_library_conditional_alternative.xml]: feature flag added

File: forms/preside-objects/email_content_library_content.xml (hunk 6 above)
Comment (seakchiew) [reply forms/preside-objects/email_content_library_content.xml]: feature flag added

**Comment by seakchiew** [comment]
hi @DominicWatson 

I think i missed the rendering part for email content and have updated the rendering part. Could you help to revieew this PR?

Thanks
Seak Chiew

**Pull request merged**